internal/utf16/string_test.go

```diff
@@ -0,0 +1,79 @@
+package utf16_test
+
+import (
+	"testing"
+
+	"github.com/NeowayLabs/abad/internal/utf16"
+)
+
+var S = utf16.S
+
+func TestStrIndex(t *testing.T) {
+	for _, tc := range []struct {
+		str      utf16.Str
+		sub      utf16.Str
+		index int
+		contains bool
+	}{
+		{
+			str:      S("hello world"),
+			sub:      S("h"),
+			index: 0,
+			contains: true,
+		},
+		{
+			str:      S("hello world"),
+			sub:      S("e"),
+			index: 1,
+			contains: true,
+		},
+		{
+			str:      S("hello world"),
+			sub:      S("l"),
+			index: 2,
+			contains: true,
+		},
+		{
+			str:      S("hello world"),
+			sub:      S(""),
+			index: 0,
+			contains: true,
+		},
+		{
+			str:      S("hello world"),
+			sub:      S("world"),
+			index: 6,
+			contains: true,
+		},
+		{
+			str:      S(""),
+			sub:      S("world"),
+			index: -1,
+			contains: false,
+		},
+		{
+			str:      S("hello"),
+			sub:      S("hello "),
+			index: -1,
+			contains: false,
```

Comment:
if -1 already means absent do we need a contains boolean too ?

Reply:
yes, we can!